File: elixir/anagram/anagram.exs
defmodule Anagram do
  def match(word, candidates) do
    Enum.filter(candidates, fn(x) -> 
      words_match?(String.downcase(word), String.downcase(x)) 
    end)
  end

  defp words_match?(word, word), do: false

  defp words_match?(word, another_word) do
    normalize(word) == normalize(another_word) 
  end

  defp normalize(word) do
    word 
      |> String.to_char_list 
      |> Enum.sort
  end
end

File: ruby/anagram/anagram.rb
class Anagram < String
  def match(words)
    words.select do |word| 
      !identical?(word) && anagram?(word)
    end
  end

  private

  def identical?(word)
    word.downcase == downcase
  end

  def anagram?(word)
    word.downcase.chars.sort == downcase.chars.sort 
  end
end
